### PR TITLE
option to auto-lock database after an auto-type action

### PR DESCRIPTION
**Pull request by lawrencium**
Pull Request for https://github.com/keeweb/keeweb/issues/587

Pretty much followed the structure of locking on minimizing & copy password. 

The option defaults to false (auto-type does not trigger lock unless user explicitly checks the box).

### CODE REVIEW

File: app/scripts/auto-type/index.js [hunk 1]
```diff
@@ -143,6 +144,10 @@ const AutoType = {
         } else {
             callback();
         }
+
+        if (AppSettingsModel.instance.get('lockOnAutoType')) {
```
Comment (antelle) [inline app/scripts/auto-type/index.js]: Please move to `runAndHandleResult`

File: app/templates/settings/settings-general.hbs [hunk 2]
```diff
@@ -147,6 +147,12 @@
         <label for="settings__general-lock-on-copy">{{res 'setGenLockCopy'}}</label>
     </div>
 
+    <div>
```
Comment (antelle) [inline app/templates/settings/settings-general.hbs]: This must not be visible on web. Please wrap with `{{#if canAutoType}}` and use `AutoType.enabled` for it.

**Comment by antelle** [comment]
I just tested this, and `auto-lock on minimize` locks the file for me. So, the suggestion is to rename the option to 'minimize or auto-type'. What happens for you, which OS are you using?

**Comment by lawrencium** [comment]
I'm on Mac OS X. I played around w/ auto-lock-on minimize, and I found it kind of confusing to think about, primarily (I think) due to the changes brought about by #497. 

The `auto-lock on minimize` option locks on auto-type if the window the auto-type is performed on completely overlaps the KeeWeb app. However, if you don't overlap the KeeWeb app and the window you are auto-typing on, then KeeWeb will not lock because it won't consider itself hidden. So, it is possible to have `auto-lock on minimize` enabled, complete an auto-type, and not have the database lock itself afterwards.

I'm not a big fan of #497 because I think it overloads the definition of "minimize" and I think there should be a separate option to `auto-lock on hide`.

**Comment by antelle** [comment]
Yes, probably it makes sense to revert that change, and then it becomes clear.

**Comment by lawrencium** [comment]
So, after playing around w/ the electron framework and reading #497, I think that the implementation of #497 (done in #575) isn't correct. An event listener for 'hide' is triggered whenever the window _is not visible_, meaning that a 'hide' event is emitted whenever the app cannot be seen. This is different than the intent of #497, which is looking to bind a Mac-specific type of event to 'minimize'.

I attached a commit to revert the changes in #575. I think the best way to handle #497 is to submit a pull request to the electron repo to either 1) emit a new event when a hide-action is performed or 2) also emit a 'minimize' event when a hide-action is performed.

Do you guys have thoughts about this?

**Comment by antelle** [comment]
I absolutely agree, we can't treat automatic hide as a user action, so it's better to revert that commit.

**Comment by lawrencium** [comment]
Thanks, so do you think this pull request can be merged?

**Comment by antelle** [comment]
yes, of course, I'll merge it tonight

**Comment by antelle** [comment]
Merged to develop, uploaded to beta

**Pull request merged**